packages/stark-ui/src/modules/transform-input-directive/directives/transform-input.directive.spec.ts

```diff
@@ -0,0 +1,285 @@
+/*tslint:disable:completed-docs no-identical-functions no-duplicate-string no-big-function*/
+import { Component } from "@angular/core";
+import { ComponentFixture, TestBed } from "@angular/core/testing";
+import { FormControl, FormsModule, ReactiveFormsModule } from "@angular/forms";
+import { StarkInputTransformationType, StarkTransformInputDirective } from "./transform-input.directive";
+import { Observer } from "rxjs";
+
+/**
+ * Mocks an InputEvent on the element with the given value.
+ * @param value - the value to set the element to.
+ * @param element - the HTMLInputElement to mock the event on
+ */
+const mockInputEvent: (value: string, element: HTMLInputElement) => void = (value: string, element: HTMLInputElement): void => {
+	const inputEvent: Event = document.createEvent("Event");
+	inputEvent.initEvent("input", true, true);
+
+	element.value = value;
+	element.dispatchEvent(inputEvent);
+};
+
+describe("TransformInputDirective", () => {
```

Comment:
Could you add a test for the error thrown when an invalid transformation is provided?

Reply:
For each implementation (formControl, ngModel, nothing), or once?